### PR TITLE
Fix node lineage in JexlNodeFactory.createNodeTreeFromFieldsToValues

### PR DESCRIPTION
**Pull request by apmoriarty**
This is a second cut of https://github.com/NationalSecurityAgency/datawave/pull/1273 after the changes to refactor the parallel index expansion.

Change summary
- instead of building out nodes in-place, children are built first and the parent node is build last (simplifies logic)
- more instances of copying a node (preserves node lineage)
- added tests for JexlNodeFactory and UnfieldedIndexExpansionVisitor

### CODE REVIEW

File: warehouse/query-core/src/test/java/datawave/query/jexl/JexlNodeFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    
+    // It's worth noting that 'expandFields' is almost always set to true. This test is a contrived case that exercises a code path.
+    @Test
+    public void testNoFieldExpansion() throws ParseException {
+        IndexLookupMap indexLookupMap = new IndexLookupMap(10, 10);
+        indexLookupMap.putAll("FOO", Sets.newHashSet("6", "9"));
+        indexLookupMap.putAll("BAR", Sets.newHashSet("7", "8"));
+        
+        JexlNode originalNode = JexlASTHelper.parseJexlQuery("((_Bounded_ = true) && (FOO > 5 && FOO < 10))");
+        JexlNode node = JexlNodeFactory.createNodeTreeFromFieldsToValues(JexlNodeFactory.ContainerType.OR_NODE, false, originalNode, indexLookupMap, false,
```
Comment (avgAGB) [inline warehouse/query-core/src/test/java/datawave/query/jexl/JexlNodeFactoryTest.java:31]: Just curious, would there be a practical way to throw in a unit test that goes through the cases where ContainerType.AND_NODE is passed to the createNode method?

Comment (apmoriarty) [reply warehouse/query-core/src/test/java/datawave/query/jexl/JexlNodeFactoryTest.java:31]: Added test in https://github.com/NationalSecurityAgency/datawave/pull/1383/commits/753ec82e172a2f8ecce0be3fe7e14ee5705f78f2

Comment (avgAGB) [reply warehouse/query-core/src/test/java/datawave/query/jexl/JexlNodeFactoryTest.java:31]: Cool, thank you.

**Review by avgAGB — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BoundedRangeIndexExpansionVisitor.java [hunk 2]
```diff
@@ -105,7 +103,7 @@ protected void rebuildFutureJexlNode(FutureJexlNode futureJexlNode) {
         JexlNode currentNode = futureJexlNode.getOrigNode();
         IndexLookupMap fieldsToTerms = futureJexlNode.getLookup().lookup();
         
-        futureJexlNode.setRebuiltNode(JexlNodeFactory.createNodeTreeFromFieldsToValues(JexlNodeFactory.ContainerType.OR_NODE, new ASTEQNode(
-                        ParserTreeConstants.JJTEQNODE), currentNode, fieldsToTerms, expandFields, expandValues, futureJexlNode.isKeepOriginalNode()));
+        futureJexlNode.setRebuiltNode(JexlNodeFactory.createNodeTreeFromFieldsToValues(JexlNodeFactory.ContainerType.OR_NODE,
+                        futureJexlNode.isKeepOriginalNode(), currentNode, fieldsToTerms, expandFields, expandValues, futureJexlNode.isKeepOriginalNode()));
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BoundedRangeIndexExpansionVisitor.java]: I think the first instance of futureJexlNode.isKeepOriginalNode() should simply be a "true".  That is supposed to be the isNegated flag.

**Review by ivakegg — CHANGES_REQUESTED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BoundedRangeIndexExpansionVisitor.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BoundedRangeIndexExpansionVisitor.java]: I think we want to simplify to `false` instead. The `isNegated` flag signals that negated nodes should be built and if nothing is found for the node, prune to an ASTFalseNode.

**Review by FineAndDandy — APPROVED** [review]

**Comment by FineAndDandy** [comment]
This looks consistent, but hard to tell if it's 100% correct.

**Review by avgAGB — DISMISSED** [review]

**Review by FineAndDandy — DISMISSED** [review]

**Review by ivakegg — DISMISSED** [review]

**Review by FineAndDandy — APPROVED** [review]

**Review by ivakegg — APPROVED** [review]

**Review by avgAGB — APPROVED** [review]

**Pull request merged**